03_Modules/Configuration/src/module/services.ts

```diff
@@ -73,4 +73,152 @@ export class DeviceModelService {
       return Number(itemsPerMessageGetVariablesAttributes[0].value);
     }
   }
+
+  async updateDeviceModel(
```

Comment:
I dumped this method into chatgpt and it came up with this:
```
async updateDeviceModel(
  chargingStation: any,
  stationId: string,
  timestamp: string,
): Promise<void> {
  const attributes = [
    { component: 'ChargingStation', variable: 'Model', value: chargingStation.model },
    { component: 'ChargingStation', variable: 'VendorName', value: chargingStation.vendorName },
    { component: 'Controller', variable: 'FirmwareVersion', value: chargingStation.firmwareVersion },
    { component: 'ChargingStation', variable: 'SerialNumber', value: chargingStation.serialNumber },
    { component: 'DataLink', variable: 'IMSI', value: chargingStation.modem?.imsi },
    { component: 'DataLink', variable: 'ICCID', value: chargingStation.modem?.iccid }
  ];

  const promises = attributes
    .filter(attr => attr.value !== undefined) // Filter out attributes with undefined values
    .map(attr => this._deviceModelRepository.createOrUpdateDeviceModelByStationId(
      {
        component: { name: attr.component },
        variable: { name: attr.variable },
        variableAttribute: [
          {
            type: AttributeEnumType.Actual,
            value: attr.value,
            mutability: MutabilityEnumType.ReadOnly,
            persistent: true,
            constant: true,
          },
        ],
      },
      stationId,
      timestamp,
    ));

  await Promise.all(promises); // Wait for all promises to resolve
}
```

Reply:
Would probably be nice to have this as a bulk db update.

Reply:
I can see that refactor of being good. Will adjust